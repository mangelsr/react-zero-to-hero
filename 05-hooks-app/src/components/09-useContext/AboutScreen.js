import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {

    const { user, setUser } = useContext(UserContext);

    const handleClick = () => {
        setUser({});
    };

    return (
        <div>
            <h1>About Screen</h1>
            <hr/>

            <pre>
                { JSON.stringify(user, null, 3) }
            </pre>

            <button
                onClick={handleClick}
                className="btn btn-outline-danger">
                Logout
            </button>
        </div>
    )
}
