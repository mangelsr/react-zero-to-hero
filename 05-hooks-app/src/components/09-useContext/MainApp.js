import React, { useEffect, useState } from 'react';
import { AppRouter } from './AppRouter';

import { UserContext } from './UserContext';
import './style.css';


let initialUser = JSON.parse(localStorage.getItem('user'));
if (initialUser === null) {
    initialUser = {};
}

export const MainApp = () => {

    const [user, setUser] = useState(initialUser);

    useEffect(() => {
        localStorage.setItem('user', JSON.stringify(user));
    }, [user]);

    return (
        <UserContext.Provider value={{ user, setUser }}>
            <AppRouter/>
        </UserContext.Provider>   
    )
}
