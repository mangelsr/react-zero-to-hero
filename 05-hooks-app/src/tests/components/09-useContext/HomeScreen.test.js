import '@testing-library/jest-dom';
import { mount, shallow } from 'enzyme';
import React from 'react';

import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';


describe('HomeScreen component tests', () => {

    const user = {
        name: 'Miguel',
        email: 'mangelsr25@gmail.com'
    };
    
    const wrapper = mount(
        <UserContext.Provider value={{
            user
        }} >
            <HomeScreen />
        </UserContext.Provider>
    );
   
    test('should render component', () => {
        expect(wrapper).toMatchSnapshot();
    });

});