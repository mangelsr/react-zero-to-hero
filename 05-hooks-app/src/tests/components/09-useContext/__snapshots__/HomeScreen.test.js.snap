// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen component tests should render component 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
   "name": "Miguel",
   "email": "mangelsr25@gmail.com"
}
  </pre>
</div>
`;
